lib/util/utils.js

```diff
@@ -21,5 +25,46 @@ exports.matchMemberExpression = function (name, node) {
       return true;
     }
   }
+
   return match(node.object, nameParts.length - 2);
 };
+
+/**
+ * Matches a MemberExpression by traversing the node and returns true if any of the property names match.
+ * @param nameRegex
+ * @param node
+ * @returns {Boolean}
+ */
+exports.matchRegexMemberExpression = function matchRegexMemberExpression(nameRegex, node) {
+  if (!node) {
+    return false;
+  }
+
+  if (node.type === 'MemberExpression') {
+    if (nameRegex.test(node.property.name)) {
+      return true;
+    }
+
+    return matchRegexMemberExpression(nameRegex, node.object);
```

Comment:
This is so we can have regex matches on MemberExpressions. 
Example:

     var memberExpression = esprima.parse('member.foo.prop');
     matchRegexMemberExpression(/prop/, memberExpression); //<- this would match
     matchRegexMemberExpression(/foo\.prop/, memberExpression); //<- does not match

It's more of an API decision. The alternative would be to walk through the MemberExpression, force it into a string and run the regex on that. 
Example of alternative:

         var memberExpression = esprima.parse('member.foo.prop');
         matchRegexMemberExpression(/prop/, memberExpression); // <- would match
         matchRegexMemberExpression(/foo\.prop/, memberExpression); // <- would also match

Which would you prefer?